src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+
+  const story = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        copyState = {};
+        break;
+
+      case 'addProperties':
+        copyState = { ...copyState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete copyState[key];
+        });
+        break;
+
+      default:
+        break;
+    }
```

Comment:
CODE STYLE ISSUE: You have used a `switch` statement which is a good choice for this task. However, it's better to handle the `default` case properly. In your code, the `default` case does nothing which is not a good practice. You should throw an error or at least log a warning message that an unknown action type was provided.